File: To Do 198. House Robber/src/main/Main.java
package main;

class Solution {
    public int rob(int[] nums) {
    	if (nums == null || nums.length == 0) return 0;  
        if (nums.length == 1) return nums[0]; 
        int sum2 = nums[nums.length - 1]; 
        int sum1 = Math.max(nums[nums.length - 1], nums[nums.length - 2]); 
        for (int i = nums.length - 3; i > -1; i--) {
            int sum = Math.max(nums[i] + sum2, sum1);
            sum2 = sum1; 
            sum1 = sum; 
        }
        return sum1; 
    }
}

public class Main {
	public static void main(String[] args) {
		//int []nums = {1,2,3,1};
		int []nums = {2,7,9,3,1};
		System.out.println(new Solution().rob(nums));
	}
}

File: To Do 198. House Robber/main.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size()==0)
            return 0;
        if(nums.size()==1)
            return nums[0];
        if(nums.size()==2)
            return max(nums[0],nums[1]);
        for(int i=2;i<nums.size();i++)
            nums[i]=nums[i]+ *max_element(nums.begin(),nums.begin()+i-1);
        return *max_element(nums.begin(),nums.end());
    }
};

int main(int argc, char** argv) {
    vector<int> vec = {1,2,3,1};
    cout << Solution().rob(vec) << endl;
    return 0;
}